packages/flame/lib/src/parallax.dart

```diff
@@ -235,17 +238,27 @@ enum LayerFill { height, width, none }
 /// layer moves with different velocities to give an effect of depth.
 class Parallax {
   late Vector2 baseVelocity;
-  Vector2 _size = Vector2.zero();
   late Rect _clipRect;
   final List<ParallaxLayer> layers;
 
+  Vector2? _size;
+
+  /// Do not modify this directly, since the layers won't be resized if you do
+  Vector2? get size => _size;
+  set size(Vector2? newSize) {
+    assert(newSize != null, "Don't set the size to null");
```

Comment:
If we can't accept a null size, can't we just not allow it at all?

```suggestion
  set size(Vector2 newSize) {
```

Reply:
It has to be the same type as the getter, but I'll try to refactor this away.